processor/k8sattributesprocessor/factory.go

```diff
@@ -250,3 +251,25 @@ func errWrongKeyConfig(cfg config.Processor) error {
 
 	return nil
 }
+
+func warnDeprecatedPodAssociationConfig(logger *zap.Logger, cfg config.Processor) {
+	oCfg := cfg.(*Config)
+	warnFrom := false
+	warnName := false
+	for _, assoc := range oCfg.Association {
+		if assoc.From != "" {
+			warnFrom = true
+		}
+		if assoc.Name != "" {
+			warnName = true
+		}
+	}
+
+	if warnFrom {
+		logger.Warn("pod_association.From has been deprecated in favor of pod_association.sources.From for k8sattributes processor")
```

Comment:
I think this warning is oriented around the code rather than the user-facing config and might be unclear to some. If we want o be nice, perhaps we could actually render the new config. I.e. look for all associations using the old format and they emit message akin to:

```
Deprecated pod_association configuration detected. Please replace:

pod_association:
  - ...
  
With:

pod_association:
  sources:
    - ...

Reply:
```
2022-03-22T17:19:39.089+0100    warn    k8sattributesprocessor@v0.46.0/factory.go:272   Deprecated pod_association configuration detected. Please replace:
pod_association:
- from: resource_attribute
  name: k8s.pod.uid

with

pod_association:
- sources:
  - from: resource_attribute
  - name: k8s.pod.uid   {"kind": "processor", "name": "k8sattributes/2"}
```